### PR TITLE
Add new transaction statuses and JSONB field info

### PR DESCRIPTION
**Pull request by neewy**
Signed-off-by: neewy <n.yushkevich@hotmail.com>

### Description of the Change

This PR add information about new transaction statuses and info about limitation of JSONB field.

### Benefits

Up-to-date docs.

### Possible Drawbacks 

None.

### CODE REVIEW

File: docs/source/core_concepts/glossary.rst [hunk 1]
```diff
@@ -259,12 +259,15 @@ Transaction Status Set
 ^^^^^^^^^^^^^^^^^^^^^^
 
  - NOT_RECEIVED: requested peer does not have this transaction.
- - MST_EXPIRED: this transactions is a part of MST pipeline and has expired.
+ - MST_PENDING: this transaction is a multisignature transaction which has to be signed by more keys (as requested in quorum field).
+ - MST_EXPIRED: this transaction is a multisignature transaction which is no longer valid and is going to be deleted by this peer from MST cache.
+ - ENOUGH_SIGNATURES_COLLECTED: this transaction is a multisignature transaction which has enough signatures and it going to be validated by the peer.
```
Comment (Akvinikym) [inline docs/source/core_concepts/glossary.rst]:
```suggestion
  - ENOUGH_SIGNATURES_COLLECTED: this transaction is a multisignature transaction which has enough signatures and is going to be validated by the peer.
```

File: docs/source/core_concepts/glossary.rst [hunk 2]
```diff
@@ -259,12 +259,15 @@ Transaction Status Set
 ^^^^^^^^^^^^^^^^^^^^^^
 
  - NOT_RECEIVED: requested peer does not have this transaction.
```
Comment (Akvinikym) [inline docs/source/core_concepts/glossary.rst]: Please, arrange the statuses, so they would go from "initial" to "final" ones. For example, MST_EXPIRED should be in the end along with COMMITTED or REJECTED

File: docs/source/core_concepts/glossary.rst [hunk 3]
```diff
@@ -259,12 +259,15 @@ Transaction Status Set
...
- - MST_EXPIRED: this transactions is a part of MST pipeline and has expired.
+ - MST_PENDING: this transaction is a multisignature transaction which has to be signed by more keys (as requested in quorum field).
+ - MST_EXPIRED: this transaction is a multisignature transaction which is no longer valid and is going to be deleted by this peer from MST cache.
```
Comment (Akvinikym) [inline docs/source/core_concepts/glossary.rst]: Not sure about MST cache - client may not want to know anything about it. Maybe, explain in simpler terms or even leave as "is going to be deleted"?

File: docs/source/core_concepts/er_model.rst [hunk 4]
```diff
@@ -54,7 +54,7 @@ Account
  - domain_id — identifier of domain where the account was created, references existing domain 
  - quorum — number of signatories required for creation of valid transaction from this account
  - transaction_count – counter of transactions created by this account
- - data — key-value storage for any information, related to the account
+ - data — key-value storage for any information, related to the account. Size is limited to 268.435 megabytes (PostgreSQL JSONB field).
```
Comment (Warchant) [inline docs/source/core_concepts/er_model.rst]: mention size in bytes

File: docs/source/core_concepts/glossary.rst [hunk 5]
```diff
@@ -259,12 +259,15 @@ Transaction Status Set
...
  - STATELESS_VALIDATION_FAILED: the transaction was formed with some fields, not meeting stateless validation constraints. This status is returned to a client, who formed transaction, right after the transaction was sent. It would also return the reason — what rule was violated.
  - STATELESS_VALIDATION_SUCCESS: the transaction has successfully passed stateless validation. This status is returned to a client, who formed transaction, right after the transaction was sent.
  - STATEFUL_VALIDATION_FAILED: the transaction has commands, which violate validation rules, checking state of the chain (e.g. asset balance, account permissions, etc.). It would also return the reason — what rule was violated.
  - STATEFUL_VALIDATION_SUCCESS: the transaction has successfully passed stateful validation.
  - COMMITTED: the transaction is the part of a block, which gained enough votes and is in the block store at the moment.
+ - REJECTED: this exact transaction was rejected by the peer during stateful validation step. This is required in order to prevent replay attacks.
```
Comment (Warchant) [inline docs/source/core_concepts/glossary.rst]: how is this different from stateful invalid status?

**Comment by igor-egorov** [comment]
DCO has to be updated

**Comment by neewy** [comment]
Lol, GitHub suggestions break DCO

File: docs/source/core_concepts/glossary.rst (hunk 2 above)
Comment (neewy) [reply docs/source/core_concepts/glossary.rst]: Do not agree. This is arranged with respect to this diagram, which shows transactions pipeline and statuses are respective outcomes at each validation step:

File: docs/source/core_concepts/glossary.rst [hunk 6]
```diff
@@ -259,15 +258,15 @@ Transaction Status Set
 ^^^^^^^^^^^^^^^^^^^^^^
 
  - NOT_RECEIVED: requested peer does not have this transaction.
+ - ENOUGH_SIGNATURES_COLLECTED: this transaction is a multisignature transaction which has enough signatures and is going to be validated by the peer.
```
Comment (igor-egorov) [inline docs/source/core_concepts/glossary.rst]: this ~transaction~ is a multisignature transaction

File: docs/source/core_concepts/glossary.rst [hunk 7]
```diff
@@ -259,15 +258,15 @@ Transaction Status Set
...
  - STATELESS_VALIDATION_SUCCESS: the transaction has successfully passed stateless validation. This status is returned to a client, who formed transaction, right after the transaction was sent.
  - STATEFUL_VALIDATION_FAILED: the transaction has commands, which violate validation rules, checking state of the chain (e.g. asset balance, account permissions, etc.). It would also return the reason — what rule was violated.
  - STATEFUL_VALIDATION_SUCCESS: the transaction has successfully passed stateful validation.
  - COMMITTED: the transaction is the part of a block, which gained enough votes and is in the block store at the moment.
- - REJECTED: this exact transaction was rejected by the peer during stateful validation step. This is required in order to prevent replay attacks.
+ - REJECTED: this exact transaction was rejected by the peer during stateful validation step, in previous consensus rounds. Rejected transactions' hashes are stored in `block <#block>`__ store. This is required in order to prevent `replay attacks <https://en.wikipedia.org/wiki/Replay_attack>`__.
```
Comment (igor-egorov) [inline docs/source/core_concepts/glossary.rst]: during stateful validation step ```,``` in previous consensus rounds

The comma looks redundant

File: docs/source/core_concepts/glossary.rst (hunk 6 above)
Comment (Warchant) [reply docs/source/core_concepts/glossary.rst]: This status is returned to non-mst transactions also (even with mst_enabled=false)
